library/lua/gui/widgets.lua

```diff
@@ -433,6 +438,47 @@ function Label:setText(text)
         self.frame = self.frame or {}
         self.frame.h = self:getTextHeight()
     end
+
+    self._text_dirty = true
+end
+
+function Label:update_scroll_inset()
+    if self.show_scroll_icons == nil then
+        self._show_scroll_icons = self:getTextHeight() > self.frame_body.height and 'right' or false
+    else
+        self._show_scroll_icons = self.show_scroll_icons
+    end    
+    if self._show_scroll_icons then
+        -- here self._show_scroll_icons can only be either
+        -- 'left' or any true value which we interpret as right
+        local l,t,r,b = gui.parse_inset(self.frame_inset)
+        if self._show_scroll_icons == 'left' and l <= 0 then
+            l = 1
+        elseif r <= 0 then
+            r = 1
+        end
+        self.frame_inset = {l=l,t=t,r=r,b=b}
+    end
+end
+
+function Label:render_scroll_icons(dc,rect)
+    if self:getTextHeight() > self.frame_body.height then
```

Comment:
It might be clearer if this if test were in the calling function (onRenderFrame)

Reply:
did that